lib/tasks/jenkins.rake

```diff
@@ -1,25 +1,24 @@
 begin
   require "ci/reporter/rake/minitest"
 
-  namespace :foreman do
-    task :set_test_runner do
-      ENV['TESTOPTS'] = "#{ENV['TESTOPTS']} #{Rails.root}/test/test_runner.rb"
-    end
-  end
-
   namespace :jenkins do
-    task :unit => ["jenkins:setup:minitest", 'rake:test:units', 'rake:test:lib', 'rake:test:functionals']
-    task :integration => ["jenkins:setup:minitest", 'rake:test:integration']
-    task :lib => ["jenkins:setup:minitest", 'rake:test:lib']
-    task :functionals => ["jenkins:setup:minitest", 'rake:test:functionals']
-    task :units => ["jenkins:setup:minitest", 'rake:test:units']
+    task :unit => ['jenkins:db', 'jenkins:setup:minitest',
+                   'rake:test:units', 'rake:test:lib', 'rake:test:functionals']
+    task :integration => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:integration']
+    task :lib => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:lib']
+    task :functionals => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:functionals']
+    task :units => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:units']
 
     namespace :setup do
       task :pre_ci do
         ENV["CI_REPORTS"] = 'jenkins/reports/unit/'
         gem 'ci_reporter'
       end
-      task :minitest  => [:pre_ci, "ci:setup:minitest", "foreman:set_test_runner"]
+      task :minitest  => [:pre_ci, "ci:setup:minitest"]
+    end
+
+    task :db do
+      system('rake db:migrate RAILS_ENV=test')
```

Comment:
I'll answer here to the comment on why using our own task too.
The problem of depending on  `db:test:prepare` vs our own rake task is that it works this way:
```bash
Invoke db:test:prepare`
Invoke environment
Execute environment
Execute db:test:prepare
```

However our own tasks works like this:
```bash
Invoke jenkins:db
Execute jenkins:db
...
Execute environment
```

Back to https://github.com/theforeman/foreman/pull/2870#discussion_r47495204 , in the 1st example, the environment is loaded before the db is populated. In the 2nd one by the time environment is loaded the db is ready therefore `attribute_names` is populated. I can change `db;migrate` to `db:test:prepare` though.